scripts/poolparty.lic

```diff
@@ -56,183 +56,151 @@
 # 2.01 - Fixed the library not downloading/loading issue
 # 2.02 - Moved to Elanthia Online
 # 2.03 - Fixed commas in INFO
-# 2.04 - Changed Helper module to PoolHelper to not conflict with shared library
+# 2.04 - Changed Helper module to Helper to not conflict with shared library
+# 2.05 - Code cleanup.  Hand sanity check for boxes.
 #
 # Use ;version to verify you are not running ruby 2.0 - You need 2.4+
-# Seriouesly, if your rubsy is asking you to trust scripts its too old and this wont work.
+# Seriously, if your ruby is asking you to trust scripts its too old and this wont work.
 # Update your ruby!!!!  There is an installer on the wiki
 # Tell it to ONLY install the ruby 2.5.5 NOT the lich/simu files
 # If you dont want to update ruby i suggest you not update poolparty
 #
 # If you find any bugs please report them to ???
 
-if !Script.exists? "oleani-lib"
-echo "** MISSING LIBRARY"
-echo "** So, Hold up, wait a minute let me get that lib up in it"
-echo "** Downloading Oleani-lib"
-do_client ";repo download oleani-lib.lic"
-wait_while {running?('repository')}
+begin
+  raise "Oleani is missing" unless Script.exists? "oleani-lib.lic"
+  load 'scripts/oleani-lib.lic'
+  raise "Oleani version is out of date" if Utilities::version < Gem::Version.new('0.0.8')
+rescue => error
+  echo error.message
+  echo "Downloading Oleani-lib.lic"
+  do_client(";repository download oleani-lib.lic")
+  wait_while {running?('repository')}
+  retry
 end
 
-load 'scripts/oleani-lib.lic'
+begin
+  raise "Slop command processor is missing" unless Script.exists? "slop-lib.lic"
+  load 'scripts/slop-lib.lic'
+rescue => error
+  echo error.message
+  echo "Downloading Slop-lib.lic"
+  do_client(";repository download slop-lib.lic")
+  wait_while {running?('repository')}
+  retry
+end
 
-module PoolHelper
-  def self.check_silvers
-    action = proc {|server_string|
-      if server_string =~ /^\s*Name\:|^\s*Gender\:|^\s*Normal \(Bonus\)|^\s*Strength \(STR\)\:|^\s*Constitution \(CON\)\:|^\s*Dexterity \(DEX\)\:|^\s*Agility \(AGI\)\:|^\s*Discipline \(DIS\)\:|^\s*Aura \(AUR\)\:|^\s*Logic \(LOG\)\:|^\s*Intuition \(INT\)\:|^\s*Wisdom \(WIS\)\:|^\s*Influence \(INF\)\:/
-        nil
-      elsif server_string =~ /^\s*Mana\:\s+\-?[0-9]+\s+Silver\:\s+[0-9,]+/
-        DownstreamHook.remove("poolparty_check_silvers")
-        nil
-      else
-        server_string
-      end
-    }
+if defined?(GameObj.type_data)
+  GameObj.load_data if GameObj.type_data.empty? or GameObj.type_data.nil?
+  if GameObj.type_data.nil?
+    wait_while {running?('updater')}

```

Comment:
`;repo info updater` returns zero results, what's this?

Reply:
Something that isn't there anymore.  Removed.